File: FrontEnd/lib/domain/interfaces/usecases/plants/i_create_plant_use_case.dart
/// @file i_create_plant_use_case.dart
/// @description Interfaz del caso de uso para crear una planta.
/// @module Plants
/// @layer Domain
library;

import '../../../dtos/plants/create_plant_request_dto.dart';
import '../../../entities/plant.dart';

abstract interface class ICreatePlantUseCase {
  /// Crea una nueva planta y la devuelve con su ID asignado.
  Future<Plant> execute(CreatePlantRequestDto dto);
}

File: BackEnd/src/domain/interfaces/usecases/plants/ICreatePlantUseCase.ts
/**
 * @file ICreatePlantUseCase.ts
 * @description Interfaz del caso de uso para crear una planta.
 * @module Plants
 * @layer Domain
 */

import type { CreatePlantRequestDTO } from '../../../dtos/plants/create-plant-request.dto.js';
import type { PlantResponseDTO } from '../../../dtos/plants/plant-response.dto.js';

export interface ICreatePlantUseCase {
  /**
   * @param dto — Datos de la planta a crear.
   * @param userId — Id del usuario propietario.
   * @returns Planta creada como DTO de respuesta.
   */
  execute(dto: CreatePlantRequestDTO, userId: string): Promise<PlantResponseDTO>;
}
